qingstor/sdk/utils/file_chunk.py

```diff
@@ -82,3 +87,49 @@ def __next__(self):
 
     def __iter__(self):
         return self
+
+
+class EncryptionFileChunk(FileChunk):
+    def __init__(self,fd,key):
+        FileChunk.__init__(self,fd)
+        self.key=key
+        self.mode = AES.MODE_CBC
+        m = hashlib.md5()
+        m.update(self.key)
+        self.key = m.hexdigest()
+        self.iv=Random.new().read(AES.block_size)
+
+    def read(self):
+        for (event, callbacks) in self.hooks.items():
+            for callback_function in callbacks:
+                callback_function()
+        content=self.fd.read(SEGMENT_SIZE)
+        content=self.pad(content)
+        cryptor = AES.new(self.key, self.mode, self.iv)
```

Comment:
We don't need to create `cryptor` every time read called, it can moved to `__init__`